File: accsaber-database/src/main/java/de/ixsen/accsaber/database/model/PlayerCategoryStatsPK.java
package de.ixsen.accsaber.database.model;

import java.io.Serializable;
import java.util.Objects;

public class PlayerCategoryStatsPK implements Serializable {

    protected Long player;
    protected Long category;

    public PlayerCategoryStatsPK() {
    }

    public PlayerCategoryStatsPK(Long player, Long category) {
        this.player = player;
        this.category = category;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || this.getClass() != o.getClass()) return false;
        PlayerCategoryStatsPK that = (PlayerCategoryStatsPK) o;
        return this.player.equals(that.player) && this.category.equals(that.category);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.player, this.category);
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/model/PlayerCategoryStatsPK.kt
package de.ixsen.accsaber.database.model

import java.io.Serializable
import java.util.*

class PlayerCategoryStatsPK : Serializable {
    protected var player: Long? = null
    protected var category: Long? = null

    constructor() {}
    constructor(player: Long?, category: Long?) {
        this.player = player
        this.category = category
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || this.javaClass != o.javaClass) return false
        val that = o as PlayerCategoryStatsPK
        return player == that.player && this.category == that.category
    }

    override fun hashCode(): Int {
        return Objects.hash(player, this.category)
    }
}